scripts/scil_filter_tractogram_anatomically.py

```diff
@@ -140,104 +239,82 @@ def main():
     o_dict['streamline_count_after_filtering_length_loops'] =\
         len(loop_length_sft.streamlines)
 
-    # STEP 3 - Filtering CSF endings
-    csf_labels = [4, 5, 14, 15, 24, 43, 44, 72]
-
-    # Mask creation
-    if args.csf_bin:
-        img = nib.load(args.csf_bin)
-    else:
-        img = nib.load(args.in_wmparc)
+    if args.save_outliers:
+        outlier = compute_outliers(length_sft, loop_length_sft)
+        o_dict['streamline_count_discarded_when_filtering_loops'] =\
+            len(outlier.streamlines)
+        save_outliers(out_path, 'no_loops', 'length_no_loops', in_tracto_name,
+            ext, outlier, loop_length_sft)
 
-    if not is_header_compatible(img, loop_length_sft):
-        parser.error('Headers from the tractogram and the mask are '
-                     'not compatible.')
+    if len(loop_length_sft.streamlines) == 0:
+        if args.no_empty:
+            logging.debug("The file {} won't be written (0 streamline after "
+                          "filtering loops).".format(args.out_tractogram))
+            return
+        logging.debug('The file {} contains 0 streamline after filtering '
+                      'loops'.format(args.out_tractogram))
+        save_tractogram(loop_length_sft, args.out_tractogram)
+        return
 
+    # STEP 3 - Filtering CSF endings
+    # Mask creation
     if args.csf_bin:
-        mask = get_data_as_mask(img)
+        mask = get_data_as_mask(img_csf)
     else:
-        atlas = get_data_as_label(img)
-        mask = np.zeros(atlas.shape, dtype=np.uint16)
-        for csf_lab in csf_labels:
-            id = atlas == csf_lab
-            mask[id] = 1
+        atlas = get_data_as_label(img_wmparc)
+        mask = binarize_labels(atlas, data["csf_labels"])
 
     # Filter tractogram
-    loop_length_CSF_sft , _ = filter_grid_roi(loop_length_sft, mask,
+    loop_length_csf_sft , _ = filter_grid_roi(loop_length_sft, mask,
         'any', True)
```

Comment:
PEP 8: continuation line under-indented for visual indent

Reply:
Done in dbd0b2a.